src/language-markdown/mdx.js

```diff
@@ -26,7 +26,7 @@
 
 const IMPORT_REGEX = /^import\s/;
 const EXPORT_REGEX = /^export\s/;
-const BLOCKS_REGEX = "[a-z\\.]*(\\.){0,1}[a-z][a-z0-9\\.]*";
+const BLOCKS_REGEX = "[a-z\\.]*(\\.){0,1}[a-z]?[a-z0-9\\.]*";
```

Comment:
```suggestion
const BLOCKS_REGEX = "[a-z\\.]*(\\.){0,1}[a-z][a-z0-9\\.]*|";
```
We want "a valid JSX element name OR the empty string". Your regexp allows invalid element names like `2019` to slip through. I still don't like the resulting regexp because it seems to allow strange things like `<aa...vv...>`, but for this PR my suggestion might suffice.

Reply:
@thorn0
How about this one:
```js
const BLOCKS_REGEX = "(?:[a-z])[a-z\\.]*(\\.){0,1}[a-z]?[a-z0-9\\.]*$";
```

Reply:
@thorn0 How about this one:

```js
const BLOCKS_REGEX = "([a-z]([a-z0-9]*\\.?[a-z]+[a-z0-9]*)*|)";
```

Reply:
Yes, this one is better. The outer parens aren't needed though.

Reply:
I'd like to keep it as is which seems clearer to me.

Reply:
The regexp on the next line does not have them.